Release 5.4.0 (#1919)
microsoft/onefuzz: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## 5.4.0
### Added
* Agent: Added the OneFuzz version and tool name to the Crash Report. [#1635](https://github.com/microsoft/onefuzz/pull/1635)

> * Agent: Added a check for missing libraries when running the `-help` check. [#1812](https://github.com/microsoft/onefuzz/pull/1812)

Reviewer: ```suggestion
* Agent: Added a check for missing libraries when running the LibFuzzer `-help` check. [#1812](https://github.com/microsoft/onefuzz/pull/1812)
```

* Service: Restrict node operations to administrators. [#1779](https://github.com/microsoft/onefuzz/pull/1779)

### Changed
* CLI/Service: Updated multiple first-party and third-party Python dependencies. [#1784](https://github.com/microsoft/onefuzz/pull/1784)
* Agent/Supervisor/Proxy: Updated multiple third-party Rust dependencies. [#1619](https://github.com/microsoft/onefuzz/pull/1619), [#1644](https://github.com/microsoft/onefuzz/pull/1644), [#1645](https://github.com/microsoft/onefuzz/pull/1645), [#1646](https://github.com/microsoft/onefuzz/pull/1646), [#1655](https://github.com/microsoft/onefuzz/pull/1655), [#1700](https://github.com/microsoft/onefuzz/pull/1700), [#1849](https://github.com/microsoft/onefuzz/pull/1849), [#1882](https://github.com/microsoft/onefuzz/pull/1882)
* Agent: Separate `llvm-symbolizer` setup from sanitizer environment variable initialization. [#1778](https://github.com/microsoft/onefuzz/pull/1778)
* Agent: Set the TSan options based on the external symbolizer. [#1787](https://github.com/microsoft/onefuzz/pull/1787)

> * Service: Removed the `client_secret` from the configuration file to eliminate the potential for the secret to be persisted. If you have persisted a client secret in your configuration file, remove it. [#1918](https://github.com/microsoft/onefuzz/pull/1918)

Reviewer: ```suggestion
* CLI: Added the `ONEFUZZ_CLIENT_SECRET` environment variable and removed the `client_secret` field from the configuration file. This prevents accidental misuse via persisting the secret to disk outside of [confidential client](https://docs.microsoft.com/en-us/azure/active-directory/develop/developer-glossary#client-application) environments. If you have set a client secret in your configuration file in a public client, we recommending removing and revoking it. CI scripts that currently set the client secret in the config must instead pass it via the `ONEFUZZ_CLIENT_SECRET` environment variable or on each CLI invocation via the `--client_secret` argument. [#1918](https://github.com/microsoft/onefuzz/pull/1918)
```

> * Service: Use a sas url to download log files. [#1920](https://github.com/microsoft/onefuzz/pull/1920)

Reviewer 2: ```suggestion
* CLI: Use a sas url to download log files. [#1920](https://github.com/microsoft/onefuzz/pull/1920)
```

Reviewer: ```suggestion
* CLI: Use a SAS URL to download log files. [#1920](https://github.com/microsoft/onefuzz/pull/1920)
```
(apply to any other use of acronyms like "SAS", "URL", "CLI")

### Fixed
* Agent: Only watch directories for change events. [#1859](https://github.com/microsoft/onefuzz/pull/1859)

> * Agent: Switch to a smart constructor to minimize miss-use. [#1865](https://github.com/microsoft/onefuzz/pull/1865)

Reviewer: ```suggestion
* Agent: Switch to a smart constructor to minimize misuse. [#1865](https://github.com/microsoft/onefuzz/pull/1865)
```
